v2/emailpassword/nextjs/app-directory/protecting-route.mdx

```diff
@@ -352,45 +377,85 @@ export const TryRefreshComponent = () => {
 Lets modify the Home page server component we created earlier:
 
 ```tsx title="app/components/home.tsx"
-import { getSSRSession } from "supertokens-node/nextjs";
-import { SessionContainer } from "supertokens-node/recipe/session";
-import { cookies, headers } from "next/headers";
+import { cookies } from "next/headers";
 declare let TryRefreshComponent: any; // typecheck-only, removed from output
 import { redirect } from "next/navigation";
 // @ts-ignore
 import { TryRefreshComponent } from "./tryRefreshClientComponent";
+import jwksClient from "jwks-rsa";
+import JsonWebToken from "jsonwebtoken";
+import type { JwtHeader, JwtPayload, SigningKeyCallback } from "jsonwebtoken";
 // @ts-ignore
-import { ensureSuperTokensInit } from "../config/backend";
+import { appInfo } from "../config/appInfo";
 
-ensureSuperTokensInit();
+const client = jwksClient({
+    jwksUri: `${appInfo.apiDomain}${appInfo.apiBasePath}/jwt/jwks.json`,
+});
 
-async function getSSRSessionHelper(): Promise<{ session: SessionContainer | undefined; hasToken: boolean; hasInvalidClaims: boolean, error: Error | undefined }> {
-    let session: SessionContainer | undefined;
-    let hasToken = false;
-    let hasInvalidClaims = false;
-    let error: Error | undefined = undefined;
-    
+function getAccessToken(): string | undefined {
+    return cookies().get("sAccessToken")?.value;
+}
+
+function getPublicKey(header: JwtHeader, callback: SigningKeyCallback) {
+    client.getSigningKey(header.kid, (err, key) => {
+        if (err) {
+            callback(err);
+        } else {
+            const signingKey = key?.getPublicKey();
+            callback(null, signingKey);
+        }
+    });
+}
+
+async function verifyToken(token: string): Promise<JwtPayload> {
+    return new Promise((resolve, reject) => {
+        JsonWebToken.verify(token, getPublicKey, {}, (err, decoded) => {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(decoded as JwtPayload);
+            }
+        });
+    });
+}
+
+/**
+ * A helper function to retrieve session details on the server side.
+ *
+ * NOTE: This function does not use the getSSRSession function from the supertokens-node SDK
+ * because getSession can update the access token. These updated tokens would not be
+ * propagated to the client side, as request interceptors do not run on the server side.
```

Comment:
```suggestion
 * NOTE: This function does not use the getSession / verifySession function from the supertokens-node SDK
 * because those functions may update the access token. These updated tokens would not be
 * propagated to the client side properly, as request interceptors do not run on the server side.
 * So instead, we use regular JWT verification library
```

Reply:
done